internal/indexer/moniker.go

```diff
@@ -33,14 +40,20 @@ func (i *Indexer) emitImportMoniker(sourceID uint64, p *packages.Package, obj ty
 	pkg := monikerPackage(obj)
 
 	for _, moduleName := range packagePrefixes(pkg) {
-		if moduleVersion, ok := i.dependencies[moduleName]; ok {
-			i.addMonikers(
-				"import",
-				strings.Trim(fmt.Sprintf("%s:%s", pkg, monikerIdentifier(i.packageDataCache, p, obj)), ":"),
-				sourceID,
-				i.ensurePackageInformation(moduleName, moduleVersion),
-			)
+		version, ok := i.dependencies[moduleName]
+		if ok {
+			// If this is a package reference, remove the trailing `:`
+			monikerIdentifier := fmt.Sprintf("%s:%s", pkg, monikerIdentifier(i.packageDataCache, p, obj))
+			monikerIdentifier = strings.Trim(monikerIdentifier, ":")
```

Comment:
This trimming logic is duplicated in `emitExportMoniker` and `emitImportMoniker`. Might be worth consolidating since it's easy to forget to trim `:` resulting in hanging moniker IDs.